src/app/shared/element-child-data-classes-child-elements-list/flattened-data-classes-list.component.html

```diff
@@ -0,0 +1,339 @@
+          matTooltip="Delete"
+        >
+          <span class="far fa-trash-alt"></span>
+        </button>
+      </td>
+    </ng-container>
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr
+      mat-row
+      *matRowDef="let row; columns: displayedColumns"
+      cdkDrag
+      [cdkDragData]="row"
+      [ngClass]="{ 'mdm--active-row ': row.checked }"
+    ></tr>
+  </table>
+</div>
+<div
+  class="bordered is-hidden pxy-2"
+  [ngClass]="{ block: !totalItemCount && !isLoadingResults }"
+>
+  <p class="marginless">
+    There are no Data Classes in this model. You can add one by clicking the
+    button above.
+  </p>
+  <p *ngIf="!hideFilters" class="marginless">
+    This might be due to the current filter settings.
+  </p>
+</div>
+<div *ngIf="isLoadingResults">
+  <table class="mdm--mat-table mdm--skeleton-table">
+    <tr>
+      <th
+        style="width: 5%"
+        scope="col"
+        *ngIf="isEditable && !parentDataModel.finalised"
+      >
+        <ngx-skeleton-loader
+          count="1"
+          [theme]="{ 'background-color': '#b7bbc5', width: '24px' }"
+        ></ngx-skeleton-loader>
+      </th>
+      <th style="width: 29%" scope="col">
+        <ngx-skeleton-loader
+          count="1"
+          [theme]="{ 'background-color': '#b7bbc5', width: '30%' }"
+        ></ngx-skeleton-loader>
+      </th>
+      <th style="width: 43%" scope="col">
+        <ngx-skeleton-loader
+          count="1"
+          [theme]="{ 'background-color': '#b7bbc5', width: '25%' }"
+        ></ngx-skeleton-loader>
+      </th>
+      <th style="width: 11%" scope="col">
+        <ngx-skeleton-loader
+          count="1"
+          [theme]="{ 'background-color': '#b7bbc5', width: '40%' }"
+        ></ngx-skeleton-loader>
+      </th>
+      <td
+        style="width: 2%"
+        *ngIf="isEditable && !parentDataModel.finalised"
+      ></td>
+    </tr>
+    <tr>
+      <td *ngIf="isEditable && !parentDataModel.finalised">
+        <ngx-skeleton-loader
+          count="1"
+          [theme]="{ 'background-color': '#b7bbc5', width: '100%' }"
+        ></ngx-skeleton-loader>
+      </td>
+      <td>
+        <ngx-skeleton-loader count="1" appearance="line"></ngx-skeleton-loader>
+      </td>
+      <td>
+        <ngx-skeleton-loader count="1" appearance="line"></ngx-skeleton-loader>
+      </td>
+      <td>
+        <ngx-skeleton-loader count="1" appearance="line"></ngx-skeleton-loader>
+      </td>
+      <td
+        style="display: flex; justify-content: flex-end"
+        *ngIf="isEditable && !parentDataModel.finalised"
+      >
+        <ngx-skeleton-loader
+          count="1"
+          appearance="circle"
+          [theme]="{ 'border-radius': '10px' }"
+        ></ngx-skeleton-loader>
+      </td>
+    </tr>
+  </table>
+</div>
+<div
+  class="mdm--mat-pagination"
+  [ngClass]="{ 'is-hidden': totalItemCount < 6 }"
+>
+  <mdm-paginator [length]="totalItemCount" showFirstLastButtons></mdm-paginator>
```

Comment:
The paginator is not set correctly, does not know what the current page and page size are so cannot determine whether to enable the next/previous buttons. Look at `terminology/term-list/term-list.component` for how this works (also another example of a list/table with filtering/sorting).

Reply:
fixed, also a bug related to the item count was fixed with this